design-documents/graph-ql/coverage/returns.graphqls

```diff
@@ -0,0 +1,197 @@
+input EnteredCustomAttributeInput {
+    id: ID!
+    value: String!
+}
+
+type CreateReturnOutput {
+    return: Return
+}
+
+input AddReturnCommentInput {
+    return_id: ID!
+    comment_text: String!
+}
+
+type AddReturnCommentOutput {
+    return: Return
+}
+
+input AddReturnTrackingInput {
+    return_id: ID!
+    carrier_id: ID!
+    tracking_number: String!
+}
+
+type AddReturnTrackingOutput {
+    return: Return
+}
+
+input RemoveReturnTrackingInput {
+    return_tracking_id: ID!
+}
+
+type RemoveReturnTrackingOutput {
+    return: Return
+}
+
+enum CustomAttributesPageEnum {
+    RETURN_ITEM_EDIT_FORM
+    RETURN_ITEMS_LISTING
+    # See https://github.com/magento/architecture/blob/master/design-documents/graph-ql/framework/attributes-metadata.md
+    # PRODUCTS_COMPARE
+    # PRODUCTS_LISTING
+    # ADVANCED_CATALOG_SEARCH
+}
+
+type Customer {
+    returns(
+        pageSize: Int = 20 @doc(description: "Specifies the maximum number of results to return at once. Defaults to 20."),
+        currentPage: Int = 1 @doc(description: "Specifies which page of results to return. The default value is 1."),
+    ): Returns @doc(description: "Information about the customer returns.")
+    return(id: ID!): Return @doc(description: "Get customer return details by its ID.")
+}
+
+type CustomerOrder {
+    returns(
+        pageSize: Int = 20 @doc(description: "Specifies the maximum number of results to return at once. Defaults to 20."),
+        currentPage: Int = 1 @doc(description: "Specifies which page of results to return. The default value is 1."),
+    ): Returns @doc(description: "Information about the order returns.")
+    items_eligible_for_return: [OrderItemInterface] @doc(description: "A list of order items eligible for return.")
+}
+
+type Returns {
+    items: [Return] @doc(description: "List of returns")
+    page_info: SearchResultPageInfo @doc(description: "Pagination metadata")
+    total_count: Int @doc(description: "Total count of customer returns")
+}
+
+type Return @doc(description: "Customer return") {
+    id: ID! @doc(description: "Return number")
+    order_id: String! @doc(description: "ID of the order based on which the return was created.")
+    creation_date: String! @doc(description: "The date when the return was requested.")
+    customer_email: String! @doc(description: "Email of the customer who created the return.")
+    customer_name: String @doc(description: "The name of the person returning the item")
+    status: ReturnStatus @doc(description: "Return status.")
+    shipping: ReturnShipping @doc(description: "Shipping information for the return.")
+    comments: [ReturnComment] @doc(description: "A list of comments posted for the return.")
+    items: [ReturnItem] @doc(description: "A list of items being returned.")
+    available_shipping_carriers: [ReturnShippingCarrier] @doc(description: "A list of shipping carriers available for returns.")
+}
+
+type ReturnItem {
+    id: ID!
+    product: ProductInterface!
```

Comment:
Looks like complication to me but may make sense depending on the use cases we want to support. Currently we show product name and sku.

If we would like to show selected options, discounts, quantities (like refunded, returned, shipped etc) then it makes sense to change reference to OrderItem here. @nrkapoor what do you think?

Reply:
> Looks like complication to me but may make sense depending on the use cases we want to support. Currently we show product name and sku.

I think the big value in adding edges to the graph like this is that we don't need to predict all use-cases ahead of time - client will be able to traverse and get what they need from the order line item for their use-case

Reply:
@paliarush it would be more flexible just reference OrderItem for sure. That way we can future proof the schema.

Reply:
Changed to `OrderItem`. 

To make it clear, product info will be limited to sku, sale price, name and url key, since `OrderItem` does not have reference to the product object. It should be enough to render current UI.